### PR TITLE
MM-47275 : Migrate 'header_footer_template_route.jsx' to TypeScript

### LINKED ISSUE #23407: Migrate 'header_footer_template_route.jsx' to TypeScript
We are commencing the migration of the mattermost-server/webapp to TypeScript in order to enhance code quality. This "Help Wanted" issue pertains to the migration of the aforementioned "JSX" files and their associated test files to TypeScript.

Below is a checklist outlining the tasks to be performed during the migration:

[ ] Rename the files to their corresponding TypeScript extensions (e.g., js to ts, jsx to tsx).
[ ] Update any components that utilize these files with the correct imports.
[ ] Migrate the associated tests, if applicable.
[ ] Address any errors raised by the TypeScript compiler. You can execute make check-types to view any encountered errors.

Our objective is to maintain a high level of stringency during this process. However, if you encounter any obstacles where successful migration would necessitate significant changes to other libraries or our tsconfig.json file, please raise the issues. We aim to ensure a smooth and gradual migration while enforcing as many changes as possible.

For reference, you can find examples of already migrated files in the following pull requests:
https://github.com/mattermost/mattermost-webapp/pull/3790
https://github.com/mattermost/mattermost-webapp/pull/3472
https://github.com/mattermost/mattermost-webapp/pull/4230

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-47275

### PR DESCRIPTION
**Pull request by notlelouch**
resolves #23407 by moving the contents of header_footer_template_route.jsx file to its corresponding header_footer_template_route.tsx file.

Jira https://mattermost.atlassian.net/browse/MM-47275

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/formatted_markdown_message.test.tsx
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/formatted_markdown_message.test.tsx:1]: please revert this file changes its unrelated

File: webapp/channels/src/components/header_footer_template_route.tsx
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/header_footer_template_route.tsx:1]: please change the existing file instead of creating a new file

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by M-ZubairAhmed** [comment]
@notlelouch please sign the CLA following this comment https://github.com/mattermost/mattermost/pull/23785#issuecomment-1595969683

**Comment by notlelouch** [comment]
thank you so much for your feedback, I will make the changes that you mentioned. I am sorry I am really new to open source

**Comment by M-ZubairAhmed** [comment]
> thank you so much for your feedback, I will make the changes that you mentioned. I am sorry I am really new to open source

No problem at all! It's great to see your enthusiasm and willingness. Remember, we all start somewhere. Don't hesitate to reach out whenever you have any questions. 

P.S i am a fan of Kimetsu no Yaiba too

**Pull request closed without merging**